From 0a4f4e0b69599d6eb51f96d17141a8c0331b6943 Mon Sep 17 00:00:00 2001
From: Yves-Alexis Perez <corsac@debian.org>
Date: Tue, 11 Dec 2012 21:37:00 +0100
Subject: [PATCH] Fix build when using -Wl,-z,defs

GStreamer thumbnailers should be linked against libmath and
libgsttags.

Do the first directly from Makefile.am, and implement the second in the
AC macros for GSTREAMER support.

Signed-off-by: Yves-Alexis Perez <corsac@debian.org>
---
 acinclude.m4                        |    7 +++++--
 plugins/gst-thumbnailer/Makefile.am |    5 ++++-
 2 files changed, 9 insertions(+), 3 deletions(-)

diff --git a/acinclude.m4 b/acinclude.m4
index e84b8c3..4105e37 100644
--- a/acinclude.m4
+++ b/acinclude.m4
@@ -157,8 +157,11 @@ if test x"$ac_tumbler_gstreamer_thumbnailer" = x"yes"; then
   dnl Check for gdk-pixbuf
   PKG_CHECK_MODULES([GDK_PIXBUF], [gdk-pixbuf-2.0 >= 2.14],
   [
-    dnl Check for libgstreamerthumbnailer
-    PKG_CHECK_MODULES([GSTREAMER], [gstreamer-1.0], [], [ac_tumbler_gstreamer_thumbnailer=no])
+    dnl Check for libgstreamer
+    PKG_CHECK_MODULES([GSTREAMER], [gstreamer-1.0], [
+      dnl Check for libgstreamertag
+      PKG_CHECK_MODULES([GSTREAMER_TAG], [gstreamer-tag-1.0], [], [ac_tumbler_gstreamer_thumbnailer=no])
+    ], [ac_tumbler_gstreamer_thumbnailer=no])
   ], [ac_tumbler_gstreamer_thumbnailer=no])
 fi
 
diff --git a/plugins/gst-thumbnailer/Makefile.am b/plugins/gst-thumbnailer/Makefile.am
index 87c33c3..5a0cda3 100644
--- a/plugins/gst-thumbnailer/Makefile.am
+++ b/plugins/gst-thumbnailer/Makefile.am
@@ -39,6 +39,7 @@ tumbler_gst_thumbnailer_la_CFLAGS =					\
 	-DPACKAGE_LOCALE_DIR=\"$(localedir)\"				\
 	$(GDK_PIXBUF_CFLAGS)						\
 	$(GSTREAMER_CFLAGS)						\
+	$(GSTREAMER_TAG_CFLAGS)						\
 	$(GIO_CFLAGS)							\
 	$(GLIB_CFLAGS)							\
 	$(PLATFORM_CFLAGS)						\
@@ -54,8 +55,10 @@ tumbler_gst_thumbnailer_la_LIBADD =					\
 	$(top_builddir)/tumbler/libtumbler-$(TUMBLER_VERSION_API).la	\
 	$(GDK_PIXBUF_LIBS)						\
 	$(GSTREAMER_LIBS)						\
+	$(GSTREAMER_TAG_LIBS)						\
 	$(GIO_LIBS)							\
-	$(GLIB_LIBS)
+	$(GLIB_LIBS)							\
+	-lm
 
 tumbler_gst_thumbnailer_la_DEPENDENCIES =				\
 	$(top_builddir)/tumbler/libtumbler-$(TUMBLER_VERSION_API).la
-- 
1.7.10.4

